xchange-examples/src/main/java/org/knowm/xchange/examples/blockchain/trade/BlockchainTradeDemo.java

```diff
@@ -0,0 +1,109 @@
+import lombok.SneakyThrows;
+import org.knowm.xchange.Exchange;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.trade.*;
+import org.knowm.xchange.examples.blockchain.BlockchainDemoUtils;
+import org.knowm.xchange.service.trade.TradeService;
+import org.knowm.xchange.service.trade.params.CancelOrderByCurrencyPair;
+import org.knowm.xchange.service.trade.params.TradeHistoryParamCurrencyPair;
+import org.knowm.xchange.service.trade.params.TradeHistoryParams;
+import org.knowm.xchange.service.trade.params.TradeHistoryParamsTimeSpan;
+import org.knowm.xchange.service.trade.params.orders.OpenOrdersParamCurrencyPair;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.Date;
+
+import static org.knowm.xchange.examples.blockchain.BlockchainDemoUtils.END_TIME;
+
+public class BlockchainTradeDemo {
+    private static final Exchange BLOCKCHAIN_EXCHANGE = BlockchainDemoUtils.createExchange();
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+
+    public static void main(String[] args) throws IOException {
+        System.out.println("===== TRADE SERVICE =====");
+        tradeServiceDemo();
+    }
+
+    @SneakyThrows
+    private static void tradeServiceDemo() throws IOException {
+        TradeService tradeService = BLOCKCHAIN_EXCHANGE.getTradeService();
+
+        System.out.println("===== placeLimitOrder =====");
+        LimitOrder limitOrder =
+                new LimitOrder.Builder(Order.OrderType.BID, CurrencyPair.ADA_USDT)
+                        .originalAmount(new BigDecimal("1.0"))
+                        .limitPrice(new BigDecimal("0.23"))
+                        .build();
+
+        String tradeLimitOrder = tradeService.placeLimitOrder(limitOrder);
+        System.out.println(OBJECT_MAPPER.writeValueAsString(tradeLimitOrder));
+
+        System.out.println("===== placeMarketOrder =====");
+        MarketOrder marketOrder =
+                new MarketOrder.Builder(Order.OrderType.BID, CurrencyPair.ADA_USDT)
+                        .originalAmount(new BigDecimal("1.0"))
+                        .cumulativeAmount(new BigDecimal("0.22"))
+                        .build();
+
+        String tradeMarketOrder = tradeService.placeMarketOrder(marketOrder);
+        System.out.println(OBJECT_MAPPER.writeValueAsString(tradeMarketOrder));
+
+        System.out.println("===== placeStopOrder =====");
+        StopOrder stopOrder =
+                new StopOrder.Builder(Order.OrderType.BID, CurrencyPair.ADA_USDT)
+                        .originalAmount(new BigDecimal("1.0"))
+                        .stopPrice(new BigDecimal("0.21"))
+                        .build();
+
+        String tradeStopOrder = tradeService.placeStopOrder(stopOrder);
+        System.out.println(OBJECT_MAPPER.writeValueAsString(tradeStopOrder));
+
+        Thread.sleep(5000);
```

Comment:
why do we need this hardcoded sleep?

Reply:
Because it takes a few seconds to impact the order and if I don't set the delay it does not show the previously created order.